### ISSUE TITLE
Screen Mirrored is Inverted

### ISSUE THREAD

**Issue by kushhh01** [Genymobile/scrcpy#1380]
- [ yes ] I have read the [FAQ](https://github.com/Genymobile/scrcpy/blob/master/FAQ.md).
 - [ yes ] I have searched in existing [issues](https://github.com/Genymobile/scrcpy/issues).

**Environment**
 - OS: [e.g. Debian (Kali Linux),Windows]
 - scrcpy version: [e.g. 1.13]
 - installation method: [e.g. apt, Windows release...]
 - device model: 
 - Android version: [e.g. 10]

**Describe the bug**

On errors, please provide the output of the console (and `adb logcat` if relevant).

```
Please paste terminal output in a code block.
INFO: scrcpy 1.12.1 <https://github.com/Genymobile/scrcpy>
/usr/share/scrcpy/scrcpy-server: 1 fil...shed. 0.0 MB/s (24773 bytes in 1.200s)
error: more than one device/emulator
ERROR: "adb reverse" returned with value 1
WARN: 'adb reverse' failed, fallback to 'adb forward'
INFO: Initial texture: 1600x896

```

Please do not post screenshots of your terminal, just post the content as text instead.
When ever I screen mirror any device it gets mirror but inverted like every thing is inverted.

**Comment by rom1v** [comment]
Inverted how? Vertically?

If you record with:
```
adb shell screen record /sdcard/file.mp4
adb pull /sdcard/file.mp4
```

Is the video "inverted" too?

**Comment by kushu01** [comment]
Yes the video too is inverted

**Comment by rom1v** [comment]
Is it flipped vertically, or rotated 180°?

If you rotate it twice with `Ctrl`+`→` (or `scrcpy --rotation 2`), does it fix the problem?

**Comment by kushu01** [comment]
It is not flipped it is inverted

**Comment by rom1v** [comment]
What do you mean by inverted, then? Cold you post a screenshot?

**Comment by kushu01** [comment]
Yes wait I'll email you the screenshot

**Comment by kushu01** [comment]
Here is screenshot of how actually inverted it is.

**Comment by rom1v** [comment]
You can't send it by email that way, you have to post it on the GitHub thread.

**Comment by kushu01** [comment]
![Screenshot from 2020-05-11 10-35-18](https://user-images.githubusercontent.com/65150039/81544936-c7387e80-9395-11ea-8a89-593435072c05.png)

**Comment by rom1v** [comment]
OK so it's flipped (horizontally). This is a bug on your device, and scrcpy does not provide any option to workaround that.

You could force a flip by modifying the source code and building:

```diff
diff --git a/app/src/screen.c b/app/src/screen.c
index 0af8de8..14c478e 100644
--- a/app/src/screen.c
+++ b/app/src/screen.c
@@ -444,7 +444,7 @@ screen_update_frame(struct screen *screen, struct video_buffer *vb) {
 void
 screen_render(struct screen *screen) {
     SDL_RenderClear(screen->renderer);
-    if (screen->rotation == 0) {
+    if (false && screen->rotation == 0) {
         SDL_RenderCopy(screen->renderer, screen->texture, NULL, NULL);
     } else {
         // rotation in RenderCopyEx() is clockwise, while screen->rotation is
@@ -464,7 +464,7 @@ screen_render(struct screen *screen) {
         }
 
         SDL_RenderCopyEx(screen->renderer, screen->texture, NULL, dstrect,
-                         angle, NULL, 0);
+                         angle, NULL, SDL_FLIP_HORIZONTAL);
     }
     SDL_RenderPresent(screen->renderer);
 }
```

**Comment by kushu01** [comment]
I have installed it using apt. I haven't built this. So in order to build where exactly can I change source code?

**Comment by pokekrishna** [comment]
I wanted to view my screen flipped for a video-recording studio. This issue has helped me instead. Going to rebuild on Windows.

Where can I see what are the other options instead of  `SDL_FLIP_HORIZONTAL`?

**Comment by pokekrishna** [comment]
We can keep a command-line argument for showing the flipped video (both horizontal and vertical), much like other switches.
Food for thought

**Comment by mostm** [comment]
In case someone still needs this, here is a Windows build of the current commit on `master` w/ changes suggested by @rom1v 
[scrcpy-v1.21-2-g5704ec6.zip](https://drive.google.com/file/d/1LNAoyYg3GJyFFsscerGqq5G4_NsbskA3/view?usp=sharing)

**Comment by DRSDavidSoft** [comment]
@rom1v a commandline `--flip [v,h]` is appreciated

**Comment by FinlayDaG33k** [comment]
> In case someone still needs this, here is a Windows build of the current commit on `master` w/ changes suggested by @rom1v [scrcpy-v1.21-2-g5704ec6.zip](https://drive.google.com/file/d/1LNAoyYg3GJyFFsscerGqq5G4_NsbskA3/view?usp=sharing)

Build sadly doesn't work, always returns "could not execute adb push" (whereas the `1.21` release from here works just fine... with the image flipped horizontally).

**Comment by jz1** [comment]
set bluestacks graphic engine mode to "Compatibility", it will show normal display then

**Comment by gac123** [comment]
> In case someone still needs this, here is a Windows build of the current commit on `master` w/ changes suggested by @rom1v [scrcpy-v1.21-2-g5704ec6.zip](https://drive.google.com/file/d/1LNAoyYg3GJyFFsscerGqq5G4_NsbskA3/view?usp=sharing)

hey mostm, I built it as well with those changes but get error when running scrcpy (rasp pi 400, debian bullseye) ERROR: Could not initialize SDL: displayIndex must be in the range 0 - -1

were there any other changes required? or would something else be required for v1.24?

**Comment by rom1v** [comment]
Please test #4441.

**Comment by rom1v** [comment]
Implemented in v2.3.
